core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -211,7 +212,16 @@ private[spark] class ExecutorAllocationManager(
     listenerBus.addListener(listener)
 
     val scheduleTask = new Runnable() {
-      override def run(): Unit = Utils.logUncaughtExceptions(schedule())
+      override def run(): Unit = {
```

Comment:
Right, but, according to the JIRA you are dealing with an exception case. How does this stop the thread from stopping?